pkg/datapath/linux/node_linux_test.go

```diff
@@ -946,6 +946,11 @@ func (s *linuxPrivilegedIPv4OnlyTestSuite) TestArpPingHandling(c *check.C) {
 	runtime.LockOSThread()
 	defer runtime.UnlockOSThread()
 
+	prevEnableL2NeighDiscovery := option.Config.EnableL2NeighDiscovery
+	defer func() { option.Config.EnableL2NeighDiscovery = prevEnableL2NeighDiscovery }()
+
+	option.Config.EnableL2NeighDiscovery = true
```

Comment:
Enabling node port @L1014 below should've enabled neighbor discovery - https://github.com/cilium/cilium/blob/0a4a393d6554da91d34236bcd2cd0f7b6046f3a4/pkg/datapath/linux/node.go#L1378-L1378?

Reply:
If `option.Config.EnableL2NeighDiscovery` is `false` the following will happen:
```golang
		switch {
		case !option.Config.EnableL2NeighDiscovery:
			n.enableNeighDiscovery = false
		case option.Config.EnableNodePort:
```
```golang
		switch {
		case !false:
			n.enableNeighDiscovery = false
		case option.Config.EnableNodePort:
```
```golang
		switch {
		case true:
			n.enableNeighDiscovery = false <---
		case option.Config.EnableNodePort:
```

Reply:
Yeah, makes sense. I missed the ordering of switch cases.